### PR TITLE
Added test case for Uplift Model

### LINKED ISSUE #437: Add basic tests for uplift model

### PR DESCRIPTION
**Pull request by kanurag94**
Fix #437 

Please ignore closed PRs.

### CODE REVIEW

**Comment by marco-c** [comment]
The test you are adding is failing. Did you try running it?

**Comment by kanurag94** [comment]
@marco-c  The tests are running locally, I guess we have to update test DB first for taskcluster to return true?

**Comment by kanurag94** [comment]
@marco-c these commits include updated bugs.json and test cases for uplift model. They are still failing on taskcluster but runs well on local machine.
Kindly help

**Comment by kanurag94** [comment]
image
@marco-c If you can please merge it, it'll work

**Comment by chidauri** [comment]
> @marco-c these commits include updated bugs.json and test cases for uplift model. They are still failing on taskcluster but runs well on local machine.
> Kindly help

try running this, maybe other tests are broken
`python3 -m pytest tests/test_*`

**Comment by marco-c** [comment]
@kanurag94 you can see the logs on Taskcluster to check what is failing.

**Comment by kanurag94** [comment]
> > @marco-c these commits include updated bugs.json and test cases for uplift model. They are still failing on taskcluster but runs well on local machine.
> > Kindly help
> 
> try running this, maybe other tests are broken
> `python3 -m pytest tests/test_*`

Thanks!
It passes those tests well, but test_run is failing which may be the cause.

**Comment by kanurag94** [comment]
> @kanurag94 you can see the logs on Taskcluster to check what is failing.

@marco-c It seems the failure is due to test_run.py which is not included in the commit. Views?
image

**Comment by marco-c** [comment]
`test_run.py` is using the bugs.json mock file, so the failure is probably due to your changes in that file.

Two ways to fix this:
1. You need to add more bugs to the mock file, so that the model trained by test_run.py is a bit better and manages to find bugs with 0.9 confidence;
2. You add a check at https://github.com/mozilla/bugbug/blob/master/bugbug/model.py#L286 and not run https://github.com/mozilla/bugbug/blob/master/bugbug/model.py#L286-L290 if len(y_test_filter) is 0.

**Comment by kanurag94** [comment]
@marco-c I went with 2nd way and it seems the errors remain the same. Does .csv files needs to be updated after changes in bugs.json? if

**Comment by marco-c** [comment]
Are you sure the errors are the same?

**Comment by marco-c** [comment]
Looking at the logs on Taskcluster, I see a different error being reported now.

File: bugbug/model.py [hunk 1]
```diff
@@ -283,11 +283,12 @@ def train(self, importance_cutoff=0.15):
...
-                metrics.confusion_matrix(
-                    y_test_filter, y_pred_filter, labels=class_names
+            if len(y_test_filter)!=0:
+                print(
+                    metrics.confusion_matrix(
+                        y_test_filter, y_pred_filter, labels=class_names
+                    )
                 )
-            )
             print(
```
Comment (marco-c) [inline bugbug/model.py]: Move this in the if branch too to fix the other error that is showing up now.

**Comment by marco-c** [comment]
You will also need to fix the linting issues, you can use pre-commit for that.

**Comment by kanurag94** [comment]
@marco-c  You are awesome. Thanks!

**Pull request merged**